Tests/Remora.Discord.Tests/JsonAssertOptions.cs

```diff
@@ -65,6 +65,17 @@ public JsonAssertOptions
             this.AllowSkip = allowSkip ?? (_ => false);
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="JsonAssertOptions"/> class.
+        /// </summary>
+        /// <param name="original">An existing <see cref="JsonAssertOptions"/> object to be cloned..</param>
+        public JsonAssertOptions(JsonAssertOptions original)
```

Comment:
Where is this used? I can't find it.

Reply:
[Here](https://github.com/Nihlus/Remora.Discord/pull/67/files#diff-edc743fd401643d2052fa7515fd1975bf56a33fd4c40f535d1579168f5eaba01R37) for example.

Reply:
A more appropriate way of doing it might be to make this a record instead of a class - then you'd avoid needing to declare `init` properties and a handwritten copy constructor. Thoughts?

Reply:
That would work. It's the same thing, just slightly less verbose. Personally, I don't see records as the proper solution for data modeling, unless you actually WANT all the value-equality semantics they add. Copy constructors is literally how `with` syntax works, under the hood. I've just been using the `new(original)` pattern just as a placeholder until `with` syntax comes to all types.

Reply:
Value semantics are more or less appropriate for an options type. Want to do that in this PR? It's a bit more robust and has less of a maintenance overhead than the proposed changes.

Reply:
Are there more classes you'd want to see changed over to records than just this one? If so, itemize them in an issue and I'd be happy to knock them out.